.pre-commit-config.yaml

```diff
@@ -1,32 +1,26 @@
 exclude: '.*\.(pcap|pcapng|dat|txt)'
 fail_fast: false
 repos:
-  - repo: local
-    hooks:
-      - id: check-clang-format-version
-        name: Check clang-format version
-        entry: python3 ./ci/check-clang-format-version.py
-        language: system
   - repo: https://github.com/pre-commit/pre-commit-hooks
     rev: v4.6.0
     hooks:
-      - id: trailing-whitespace
       - id: check-case-conflict
+      - id: check-executables-have-shebangs
+      - id: check-json
+      - id: check-shebang-scripts-are-executable
+      - id: check-merge-conflict
+      - id: check-symlinks
+      - id: destroyed-symlinks
+      - id: detect-private-key
       - id: end-of-file-fixer
+      - id: forbid-submodules
       - id: mixed-line-ending
         args: ['--fix=lf']
+      - id: trailing-whitespace
   - repo: https://github.com/psf/black
     rev: 24.4.2
     hooks:
       - id: black
-  - repo: https://github.com/pocc/pre-commit-hooks
-    rev: v1.3.5
-    hooks:
-      - id: clang-format
-        args: ["--style=file"]  # Use the .clang-format file for configuration
-        files: ^(Common\+\+|Packet\+\+|Pcap\+\+|Tests|Examples)/.*\.(cpp|h)$
```

Comment:
But now if an user run `pre-commit --all`, he will not get the error from clang-format?

Reply:
Unfortunately yes, but pre-commit.ci has better performance. It is a drawback unfortunately

Reply:
It's okay to remove it from pre-commit. Perhaps we could provide a pre-check script for the users to run pre-commit, formatting checks, and any other checks before they commit.

Reply:
Agreed 👍. It is better to have a script for them. Still trying different things. When everything is settled, I can convert them to scripts